src/Symfony/Component/Messenger/MessageBus.php

```diff
@@ -39,6 +39,10 @@ public function __construct(iterable $middlewareHandlers = array())
      */
     public function dispatch($message)
     {
+        if (!\is_object($message)) {
+            throw new \InvalidArgumentException(sprintf('Invalid type for message argument. Expected object, but got "%s".', \gettype($message)));
```

Comment:
Can you use an `InvalidArgumentException` coming from the Messenger package? (which will also implement Messenger's `ExceptionInterface`)

Reply:
Done.